src/components/application_manager/test/commands/mobile/put_file_test.cc

```diff
@@ -108,6 +156,258 @@ TEST_F(PutFileResponceTest, Run_ApplicationRegistered_Success) {
+  (*msg_)[am::strings::msg_params][am::strings::system_file] = true;
+
+  PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
+
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+  EXPECT_CALL(mock_policy_handler_,
+              ReceiveMessageFromSDK(kFileName, binary_data_))
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  const std::string storage_folder = "/storage";
+  EXPECT_CALL(mock_app_manager_settings_, system_files_path())
+      .WillOnce(ReturnRef(storage_folder));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::GENERIC_ERROR,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(PutFileRequestTest, Run_IvalidUpdateFile_UNSUCCESS) {
+  (*msg_)[am::strings::msg_params][am::strings::offset] = kZeroOffset;
+  (*msg_)[am::strings::msg_params][am::strings::system_file] = false;
+
+  PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
+
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+  EXPECT_CALL(mock_policy_handler_,
+              ReceiveMessageFromSDK(kFileName, binary_data_))
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  EXPECT_CALL(mock_app_manager_settings_, app_storage_folder())
+      .WillOnce(ReturnRef(kStorageFolder));
+  EXPECT_CALL(*mock_app_, folder_name()).WillOnce(Return(kAppFolder));
+  const uint32_t avaliable_space = 2u;
+  EXPECT_CALL(*mock_app_, GetAvailableDiskSpace())
+      .WillOnce(Return(avaliable_space))
+      .WillOnce(Return(avaliable_space));
+
+  const std::string file_path =
+      file_system::ConcatPath(kStorageFolder, kAppFolder);
+  EXPECT_CALL(mock_app_manager_,
+              SaveBinary(binary_data_, file_path, kFileName, kZeroOffset))
+      .WillOnce(Return(mobile_apis::Result::SUCCESS));
+
+  EXPECT_CALL(*mock_app_, AddFile(_)).WillOnce(Return(false));
+  EXPECT_CALL(*mock_app_, UpdateFile(_)).WillOnce(Return(false));
+
+  MessageSharedPtr result_msg(CatchMobileCommandResult(CallRun(*command)));
+  EXPECT_EQ(mobile_apis::Result::INVALID_DATA,
+            static_cast<mobile_apis::Result::eType>(
+                (*result_msg)[am::strings::msg_params][am::strings::result_code]
+                    .asInt()));
+}
+
+TEST_F(PutFileRequestTest, Run_AddFile_SUCCESS) {
+  (*msg_)[am::strings::msg_params][am::strings::offset] = kZeroOffset;
+  (*msg_)[am::strings::msg_params][am::strings::system_file] = false;
+
+  PutFileRequestPtr command(CreateCommand<PutFileRequest>(msg_));
+
+  EXPECT_CALL(*mock_app_, hmi_level())
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+  EXPECT_CALL(mock_policy_handler_,
+              ReceiveMessageFromSDK(kFileName, binary_data_))
+      .WillOnce(Return(mobile_apis::HMILevel::HMI_FULL));
+
```

Comment:
@okozlovlux I see common part this and in 2 tests below. Please move this common part to separate method.